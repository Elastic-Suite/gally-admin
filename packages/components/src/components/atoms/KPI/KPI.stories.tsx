import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'

import KPI from './KPI'

export default {
  title: 'Atoms/KPI',
  component: KPI,
} as ComponentMeta<typeof KPI>

const Template: ComponentStory<typeof KPI> = (args) => <KPI {...args} />

export const Default = Template.bind({})

Default.args = {
  label: 'searches',
  value: 68116,
}
