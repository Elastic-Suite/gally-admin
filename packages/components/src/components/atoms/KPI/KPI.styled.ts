import { CSSProperties } from 'react'
import { styled } from '@mui/system'

export const comonStyle: CSSProperties = {
  textAlign: 'center',
  margin: 0,
  fontFamily: 'var(--gally-font)',
}

export const Title = styled('h3')(({ theme }) => ({
  ...comonStyle,
  fontSize: theme.spacing(2),
  fontWeight: 400,
  color: theme.palette.colors.neutral[600],
  textTransform: 'capitalize',
  whiteSpace: 'normal',
}))

export const Content = styled('p')(({ theme }) => ({
  ...comonStyle,
  fontSize: theme.spacing(6),
  fontWeight: 700,
  color: theme.palette.colors.primary.main,
  letterSpacing: '-0.4px',
}))

export const Container = styled('div')(({ theme }) => ({
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  justifyContent: 'center',
  gap: theme.spacing(1),
  width: 232,
}))
