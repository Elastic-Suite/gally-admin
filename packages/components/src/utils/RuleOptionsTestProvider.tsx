import React, { ReactNode } from 'react'
import {
  IRuleEngineOperators,
  RuleAttributeType,
  getIri,
} from '@elastic-suite/gally-admin-shared'

import ruleEngineOperator from '../../public/mocks/rule_engine_operators.json'
import sourceFields from '../../public/mocks/source_fields.json'
import sourceFieldLabels from '../../public/mocks/source_field_labels.json'

import RuleOptionsProvider from '../components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(getIri('source_fields', field.id))
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator as IRuleEngineOperators}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
