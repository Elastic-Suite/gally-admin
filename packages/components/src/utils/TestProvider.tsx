import React, { ReactNode } from 'react'

import catalogs from '../../public/mocks/catalog.json'
import metadata from '../../public/mocks/metadata.json'

import { catalogContext, metadataContext } from '../contexts'
import { useCatalogs } from '../hooks'

import OptionsProvider from '../components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  children: ReactNode
}

function TestProvider(props: IProps): JSX.Element {
  const { children } = props
  const contextValueCatalog = useCatalogs(catalogs['hydra:member'])
  const contextValueMetadata = metadata['hydra:member']

  return (
    <OptionsProvider>
      <catalogContext.Provider value={contextValueCatalog}>
        <metadataContext.Provider value={contextValueMetadata}>
          {children}
        </metadataContext.Provider>
      </catalogContext.Provider>
    </OptionsProvider>
  )
}

export default TestProvider
