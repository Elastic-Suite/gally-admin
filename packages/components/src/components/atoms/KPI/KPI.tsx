import React from 'react'
import { Container, Content, Title } from './KPI.styled'

export interface IKPIPros {
  label: string
  value: number
  isPercentage?: boolean
}

function KPI({ label, value, isPercentage }: IKPIPros): JSX.Element {
  return (
    <Container>
      <Title>{label}</Title>
      <Content>{`${value.toLocaleString()}${isPercentage ? '%' : ''}`}</Content>
    </Container>
  )
}

export default KPI
