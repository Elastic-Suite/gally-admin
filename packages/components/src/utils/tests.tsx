import React, { ReactNode } from 'react'
import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { Bundle, api } from '@elastic-suite/gally-admin-shared'

import { AppStore, RootState, setupStore } from '../store'

import AppProvider from '../components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: {
      'base_url/media': 'https://localhost/media/catalog/product/',
    },
    metadata: [
      {
        '@id': '/metadata/3',
        '@type': 'Metadata',
        id: 3,
        entity: 'product',
        sourceFields: ['/source_fields/217', '/source_fields/253'],
      },
      {
        '@id': '/metadata/4',
        '@type': 'Metadata',
        id: 4,
        entity: 'category',
        sourceFields: ['/source_fields/254', '/source_fields/286'],
      },
    ],
  },
  user: {
    requestedPath: '',
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
