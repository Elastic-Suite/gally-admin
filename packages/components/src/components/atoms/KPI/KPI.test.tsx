import React from 'react'

import { renderWithProviders } from '../../../utils/tests'

import KPI from './KPI'

describe('KPI match snapshot', () => {
  beforeAll(() => {
    jest
      .spyOn(Number.prototype, 'toLocaleString')
      .mockImplementation(function (this: number, _locale, options) {
        return new Intl.NumberFormat('fr-FR', options).format(this)
      })
  })

  afterAll(() => {
    jest.restoreAllMocks()
  })

  it('KPI simple', () => {
    const { container } = renderWithProviders(
      <KPI label="Test simple value" value={4998} />
    )
    expect(container).toMatchSnapshot()
  })

  it('KPI percentage', () => {
    const { container } = renderWithProviders(
      <KPI label="Test percentage value" value={49.98} isPercentage />
    )
    expect(container).toMatchSnapshot()
  })
})
