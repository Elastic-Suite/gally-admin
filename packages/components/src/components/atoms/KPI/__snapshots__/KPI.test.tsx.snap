// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KPI match snapshot KPI percentage 1`] = `
<div>
  <div
    class="css-4jtltt"
  >
    <h3
      class="css-1xva16q"
    >
      Test percentage value
    </h3>
    <p
      class="css-2384rf"
    >
      49,98%
    </p>
  </div>
</div>
`;

exports[`KPI match snapshot KPI simple 1`] = `
<div>
  <div
    class="css-4jtltt"
  >
    <h3
      class="css-1xva16q"
    >
      Test simple value
    </h3>
    <p
      class="css-2384rf"
    >
      4â€¯998
    </p>
  </div>
</div>
`;
